gapic/schema/wrappers.py

```diff
@@ -545,26 +545,14 @@ def field_headers(self) -> Sequence[str]:
     @utils.cached_property
     def flattened_fields(self) -> Mapping[str, Field]:
         """Return the signature defined for this method."""
-        answer: Dict[str, Field] = collections.OrderedDict()
         signatures = self.options.Extensions[client_pb2.method_signature]
 
-        # Iterate over each signature and add the appropriate fields.
-        for sig in signatures:
-            # Get all of the individual fields.
-            fields = collections.OrderedDict([
-                (f.strip(), self.input.get_field(*f.strip().split('.')))
-                for f in sig.split(',')
-            ])
-
-            # Sanity check: If any fields contain a message, we ignore the
-            # entire signature.
-            if any([i.message for i in fields.values()]):
-                continue
-
-            # Add the fields to the answer.
-            answer.update(fields)
+        answer: Dict[str, Field] = collections.OrderedDict(
```

Comment:
Having a hard time understanding this. If there are multiple `method_signature`s, will `answer` be a flat map of all the fields in all signatures...?

Reply:
Yes. Because python doesn't let us overload methods, we have to combine the API method signatures in the GAPIC client method signature somehow.
It's then the job of the documentation and samples to describe which combinations of flattened parameters are valid and the job of the backend service to generate errors when invalid combinations get passed.

Reply:
Cleared up AFK.